central/declarativeconfig/manager_impl.go

```diff
@@ -155,9 +161,23 @@ func (m *managerImpl) ReconcileDeclarativeConfigurations() {
 			log.Info("Start the reconciliation loop for declarative configurations")
 			m.startReconciliationLoop()
 		}
+
+		m.numberOfWatchHandlersMutex.Lock()
```

Comment:
Considering all of that is done in `m.once.Do` block, is the reason for this lock to ensure there's synchronization and this value is updated for all the threads? If so, perhaps using `atomic` would be more elegant way to do so?